pandas/core/arrays/period.py

```diff
@@ -238,6 +235,22 @@ def _generate_range(cls, start, end, periods, freq, fields):
 
         return subarr, freq
 
+    def _unbox_scalar(self, value):
+        # type: (Union[Period, NaTType]) -> int
+        if value is NaT:
+            return value.value
+        elif isinstance(value, self._scalar_type):
+            return value.ordinal
```

Comment:
does validating that the freq/dtype match matter here?

Reply:
That's handled in `_check_compatible_with`.

Currently, every use of `_unbox_scalar` follows a call to `_check_compatible_with`. We could call `_check_compatible_with` before unboxing, at the potential cost of checking twice (or additional code to avoid checking twice).

Not checking before unboxing seems dangerous.